src/pages/settings/Wallet/WalletPage/CardDetails.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import * as Expensicons from '../../../../components/Icon/Expensicons';
+import MenuItemWithTopDescription from '../../../../components/MenuItemWithTopDescription';
+import Clipboard from '../../../../libs/Clipboard';
+import useLocalize from '../../../../hooks/useLocalize';
+import usePrivatePersonalDetails from '../../../../hooks/usePrivatePersonalDetails';
+import ONYXKEYS from '../../../../ONYXKEYS';
+import _ from 'underscore';
+
+const propTypes = {
+    /** Card number */
+    pan: PropTypes.string,
+
+    /** Card expiration date */
+    expiration: PropTypes.string,
+
+    /** 3 digit code */
+    cvv: PropTypes.string,
+
+    /** User's private personal details */
+    privatePersonalDetails: PropTypes.shape({
+        /** User's home address */
+        address: PropTypes.shape({
+            street: PropTypes.string,
+            city: PropTypes.string,
+            state: PropTypes.string,
+            zip: PropTypes.string,
+            country: PropTypes.string,
+        }),
+    }),
+};
+
+const defaultProps = {
+    pan: '',
+    expiration: '',
+    cvv: '',
+    privatePersonalDetails: {
+        address: {
+            street: '',
+            street2: '',
+            city: '',
+            state: '',
+            zip: '',
+            country: '',
+        },
+    },
+};
+
+function CardDetails({pan, expiration, cvv, privatePersonalDetails}) {
+    usePrivatePersonalDetails();
+    const {translate} = useLocalize();
+
+    /**
+     * Formats an address object into an easily readable string
+     *
+     * @returns {String}
+     */
+    const getFormattedAddress = () => {
```

Comment:
I think we can move it to the card utils

Reply:
Done!